File: Iteration-1/src/FacultyTechnicalElective.java
import java.util.ArrayList;

public class FacultyTechnicalElective extends ElectiveCourse {
        private ArrayList<Course> prequisites;

        public FacultyTechnicalElective(String courseName, String courseCode, int courseCredit, int courseDay,
                        String courseHour, int quato, ArrayList<Integer> semesters,
                        ArrayList<Course> prequisites) {
                super(courseName, courseCode, courseCredit, courseDay, courseHour, quato, semesters);
                this.prequisites = prequisites;
        }

        @Override
        public boolean isEligibleToRequest(Student student) {
                return semesterControl(student) && student.getTranscript().hasBeenPassedCourses(this.getPrequisites());
        }

        //GETTER & SETTER

        public ArrayList<Course> getPrequisites() {
                return prequisites;
        }

        public void setPrequisites(ArrayList<Course> prequisites) {
                this.prequisites = prequisites;
        }
}

File: Iteration-3/Code/FacultyTechnicalElective.py
from ElectiveCourse import ElectiveCourse
from Course import Course
from typing import List

class FacultyTechnicalElective(ElectiveCourse):
    def __init__(self, courseName: str, courseCode: str, courseCredit: int, courseDay: int,
                 courseHour: str, quota: int, semesters: List[int], prerequisites: List[Course]):
        super().__init__(courseName, courseCode, courseCredit, courseDay, courseHour, quota, semesters)
        self.__prerequisites = prerequisites

    def isEligibleToRequest(self, student) -> bool:
        return self.semesterControl(student) and student.getTranscript().hasBeenPassedCourses(self.getPrerequisites()) #look at here again

    #define getter method

    def getPrerequisites(self) -> List[Course]:
        return self.__prerequisites

    # define setter method
    def setPrerequisites(self, prerequisites: List[Course]):
        self.__prerequisites = prerequisites